test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+		}
+
+		DescribeTable("Verify cgroup layout on worker node",
+			func(crun bool, ovsProc, ovsCpus, ovslb bool) {
+				if crun {
+					By("Enabling crun")
+					mcpLabel := profilecomponent.GetMachineConfigLabel(profile)
+					key, value := components.GetFirstKeyAndValue(mcpLabel)
+					mcp, err := mcps.GetByLabel(key, value)
+					Expect(err).ToNot(HaveOccurred())
+					ctrcfg := mcps.EnableCrun("cnf-crun", profile, &mcp[0])
+					testclient.Client.Create(context.TODO(), ctrcfg)
+
+					By("Applying changes in performance profile and waiting until mcp will start updating")
+					mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+					By("Waiting when mcp finishes updates")
+					mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+					defer func() {
+						By("Disabling crun")
+						testclient.Client.Delete(context.TODO(), ctrcfg)
+						By("Applying changes in performance profile and waiting until mcp will start updating")
+						mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+						By("Waiting when mcp finishes updates")
+						mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+					}()
+				}
+
+				if ovsProc {
+					result, err := chkOvsCgrpProcs(workerRTNode)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(result).To(Not(BeEmpty()))
+				}
+
+				if ovsCpus {
+					result, err := chkOvsCgrpCpuset(workerRTNode)
+					ovsCPUSet, err := cpuset.Parse(result)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(ovsCPUSet).To(Equal(onlineCPUSet))
+				}
+				if ovslb {
+					result, err := chkOvsCgroupLoadBalance(workerRTNode)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(result).To(Equal("0"))
+				}
+			},
+			Entry("[test_id:68385] Verify ovs cgroup with crun ", true, true, true, true),
+			Entry("[test_id:64098] Verify ovs cgroup with runc", false, true, true, true),
+		)
+	})
+
+	Describe("Affinity", func() {
+		Context("ovn-kubenode Pods affinity ", func() {
+			testutils.CustomBeforeAll(func() {
+				initialProfile = profile.DeepCopy()
+			})
+			It("[test_id:64100] matches with ovs process affinity", func() {
+				ovnPod, err := getOvnPod(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to get ovnPod")
+
+				ovnContainersids, err := getOvnPodContainers(&ovnPod)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Generally there are many containers inside a kubenode pods
+				// we don't need to check cpus used by all the containers
+				// we take first container
+				cpus := getCpusUsedByOvnContainer(workerRTNode, ovnContainersids[0])
+				testlog.Infof("Cpus used by ovn Containers are %s", cpus)
+				pidList, err := getOVSServicesPid(workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, pid := range pidList {
+					cmd := []string{"taskset", "-pc", pid}
```

Comment:
This code is repeated multiple times in this file, could you maybe extract that to a function?

Reply:
Addressed this in the latest commit